src/openqaoa-core/openqaoa/problems/kcolor.py

```diff
@@ -0,0 +1,194 @@
+            
+
+        Returns
+        -------
+            A random instance of the k-color problem.
+        """
+        n_nodes = kwargs.get("n_nodes", 5)
+        edge_probability = kwargs.get("edge_probability", 0.3)
+        k = kwargs.get("k", 2)
+        seed = kwargs.get("seed", None)
+        G = nx.generators.random_graphs.fast_gnp_random_graph(
+            n=n_nodes, p=edge_probability, seed=seed
+        )
+        return KColor(G, k)
+
+    @property
+    def docplex_model(self):
+        mdl = Model("k_color")
+
+        x = {(vertex, color): mdl.binary_var(name=f'x_{vertex}_{color}') for vertex in self.G for color in range(self.k)}
+
+        # Constraint: Each vertex must be assigned to exactly one color
+        for vertex in self.G:
+            mdl.add_constraint(mdl.sum(x[(vertex, color)] for color in range(self.k)) == 1)
+        cost = 0 
+        # Constraint: Adjacent vertices cannot have the same color
+        for vertex1 in self.G:
+            for vertex2 in self.G:
+                if vertex2 in self.G[vertex1]:
+                    for color in range(self.k):
+                        cost += self.penalty * x[(vertex1, color)] * x[(vertex2, color)]
+        # Objective function: Minimize the cost, finding a valid solution.
+        mdl.minimize(cost)
+        return mdl
+
+    def classical_solution(self, string: bool = False):
+        """
+        Return the classical solution of the k-color problem
+
+        Parameters
+        ----------
+        string : bool, optional
+            If the solution is returned as a string. The default is False.
+
+        Returns
+        -------
+        solution : Union[str, dict]
+            The classical solution of the specific problem as a string or a dict.
+        """
+        mdl = self.docplex_model
+        solution = mdl.solve()
+
+        if solution is None:
+            raise ValueError("No valid coloring found.")
+
+        if string:
+            coloring = ""
+        else:
+            coloring = {}
+        for var in mdl.iter_binary_vars():
+            if string:
+                coloring += str(int(round(solution.get_value(var), 1)))
+            else:
+                coloring[var.name] = int(round(solution.get_value(var), 1))
+        return coloring
+
+    @property
+    def qubo(self):
+        """
+        Returns the QUBO encoding of this problem.
+
+        Returns
+        -------
+            The QUBO encoding of this problem.
+        """
+        cplex_model = self.docplex_model
+        qubo_docplex = FromDocplex2IsingModel(cplex_model).ising_model
+
+        return QUBO(
+            self.G.number_of_nodes(),
+            qubo_docplex.terms + [[]],
+            qubo_docplex.weights + [qubo_docplex.constant],
+            self.problem_instance,
+        )
+
+    def plot_solution(self, solution: Union[dict, str], ax=None, colors=None):
+        """
+        A visualization method for the k-color solution.
```

Comment:
I noticed that this method only accepts solutions that are returned from the `classical_solution` method of this problem class.
It might be helpful to highlight that this is the case in the docstring.

I tried to pass the solution I obtained from running the actual problem through the QAOA optimization and it failed.

Maybe we should consider a method that can take the optimized outputs from running QAOA for this problem as well so we can visualize those solutions as well. What do you think @vishal-ph

Reply:
Yeah, I think that would be appropriate. Since the function is called `plot_solution` and in QAOA context the solution is indeed the bistrings, it would be seamless, if the function accepts the bitstring as input. Good point @shahidee44
Would you agree @alejomonbar?

Reply:
Sure thing! My mistake, the behavior is supposed to be what you described, just that my function to translate the bitstring didn't work .... Below is the new code to run it.

```python 

from openqaoa.problems import KColor
import matplotlib.pyplot as plt
from openqaoa import QAOA

kc = KColor.random_instance(n_nodes=5, edge_probability=0.7, k=3)
sol = kc.classical_solution()

qaoa = QAOA()
qaoa.compile(kc.qubo)
qaoa.optimize()

sol_qaoa = qaoa.result.lowest_cost_bitstrings()["solutions_bitstrings"][0]

fig, ax = plt.subplots(1, 2, figsize = (10, 5))
kc.plot_solution(sol_qaoa, ax=ax[0])
ax[0].set_title("QAOA Results")
kc.plot_solution(sol, ax=ax[1])
ax[1].set_title("CPLEX Results")
```

Reply:
Looks good! Everything works now! :)
Resolved.